### DISCUSSION TITLE
feature: stealth interaction behaviors for aggro'd NPCs

### DISCUSSION THREAD

**Discussion by Daywalkrr** [AscensionGameDev/Intersect-Engine#2413] [Ideas]
See https://github.com/AscensionGameDev/Intersect-Engine/issues/1790#issuecomment-1791902992 for high-level feature implementation details.

### Description

A player can go stealth, which makes NPCs not follow them/attack them anymore.

However, the NPCs wont attack _anyone_ during that time, unless someone successfully steals aggro back (taunt/out damaging the invisible entity).

This opens up exploits where a player can just go invisible and let their friends do all the work without anyone taking damage/aggro.

### Steps to Reproduce

1. Be in a party of 2+
2. Have someone pull aggro, then go invisible
3. Observe NPC behavior

### Version with bug

Latest

### Last version that worked well

Never

### Affected platforms

I was *not* able test on other platforms

### Duplicate Bug Check

- [X] This bug report is not a duplicate to the best of my knowledge.

**Comment by Cheshire92** [comment]
Oh yeah, I remember this one.
Probably needs a check to see whether the current target is stealthed and if so, just go for the next person on the "People that ruined my day" list.

**Comment by dhmello** [comment]
I think the check should be done, however the monster should return to its initial passive status or origin status as configured in the editor, and not look for someone to kill nearby.

And yes, this bug or anomaly continues today in beta 8

**Comment by pandinocoder** [comment]
I think that this needs to have options in the editor to configure NPC behavior.

Rationale:
- I think some games or at least some NPCs would be better by just un-targeting
- Some games/NPCs should switch to next aggro
- Some hardcore games/difficult NPCs may have stealth-breaking/bypassing mechanics
- One of the above 3 options should be configurable in the server config as the "default" (and new enum field should be nullable so that it falls back to the default if not specified)

Moving this to be a feature request based on the rationale.
